app/js/arethusa.core/services/state.js

```diff
@@ -452,6 +452,36 @@ angular.module('arethusa.core').service('state', [
       selectSurroundingToken('prev');
     };
 
+    this.getPreviousTokens = function(id,numTokens) {
```

Comment:
ah, thanks these are good catches. This is sloppy refactoring on my part. I originally coded it so that it always returned the requested number of tokens, and then realized I needed something that either returned just the previous 1 token, or ALL of the previous tokens in the sentence, so I made the numTokens parameter optional but didn't do it quite right. I will fix this. Thank you!